common/cpp/src/google_smart_card_common/admin_policy_getter.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_SMART_CARD_COMMON_ADMIN_POLICY_GETTER_H_
+#define GOOGLE_SMART_CARD_COMMON_ADMIN_POLICY_GETTER_H_
+
+#include <string>
+#include <vector>
+
+namespace google_smart_card {
+
+struct AdminPolicy {
```

Comment:
There's some code layering issue - the problem is that //common/ is reused between Smart Card Connector and third-party client applications, and we don't want to put Connector-specific details into common classes.

Just thinking out loud:
1. We have generic piece of code that sends the policy blob from the JS side to the C++ side - it's definitely abstract enough to be fine in //common/.
2. We also have the knowledge about which policies exist in Smart Card Connector - this should ideally NOT sit in //common/.
3. And we have a callsite in //third_party/pcsc-lite/naclport/server_clients_management/ that wants to check the policy.

I think the fastest option would be to move the whole admin_policy_getter.h to //third_party/pcsc-lite/naclport/server_clients_management/. That directory is Connector-specific, so it's fine to put all that code there.

Or we can think about some more future-proof solution, that'll work for policy consumers outside //third_party/pcsc-lite/naclport/server_clients_management/. One immediate option would be to change AdminPolicyGetter to only expose the policy blob as a `Value` object; but that has the downside that each callsite needs to parse the policy themselves.

Reply:
Thanks, I've moved the file. But please let me know if you would prefer the other option.

Reply:
Good for now, thanks.